src/vs/workbench/contrib/positronNotebook/browser/PositronNotebookEditorControl.ts

```diff
@@ -33,14 +33,10 @@ export class PositronNotebookEditorControl extends Disposable implements ICompos
 		super();
 
 		// Update the active code editor when the notebook selection state changes.
-		this._register(this._notebookInstance.selectionStateMachine.onNewState((state) => {
-			if (state.type === SelectionState.EditingSelection) {
-				this._activeCodeEditor = state.selectedCell.editor;
-			} else if (state.type === SelectionState.NoSelection) {
-				this._activeCodeEditor = undefined;
-			} else {
-				this._activeCodeEditor = state.selected[0]?.editor;
-			}
+		this._register(autorun(reader => {
+			const selectionStateMachine = this._notebookInstance.selectionStateMachine;
+			selectionStateMachine.state.read(reader);
+			this._activeCodeEditor = selectionStateMachine.getSelectedCells()[0]?.editor;
```

Comment:
I realize this is from my previous code but looking at it now, is this always a safe assumption that the first cell is what we want? E.g. we can always assume that the first cell selected is at the first position in the selected array and that that cell is the one we count as "selected" in terms of the editor?

Maybe a comment explaining this would be useful for the future.